src/language-js/printer-estree.js

```diff
@@ -5337,24 +5337,73 @@ function isObjectType(n) {
 }
 
 // eg; `describe("some string", (done) => {})`
-function isTestCall(n) {
+function isTestCall(n, parent) {
   const unitTestRe = /^(skip|(f|x)?(it|describe|test))$/;
+
+  if (n.arguments.length === 1) {
+    if (
+      n.callee.type === "Identifier" &&
+      n.callee.name === "async" &&
+      isTestCall(parent)
```

Comment:
Just noticed that we should make sure `parent` exists and it's a `CallExpression` before passing into `isTestCall()`, otherwise it might throw errors in some cases, e.g.

```js
function x() { async(() => {}) }
```

```
TypeError: Cannot read property 'length' of undefined
```

Reply:
@ikatyang nice catch 👍 . Though i have added a safe check to make sure parent exists and type is `CallExpression` but if i want to add a test then which file will be best to add test for this ?

Reply:
It's fine to add the test there, just need to add a comment to describe what is it used for.